frontend/src/metabase/visualizations/visualizations/ObjectDetail.jsx

```diff
@@ -25,7 +26,20 @@ const ObjectDetailProperties = {
   disableClickBehavior: true,
   settings: {
     ...columnSettings({ hidden: true }),
-    ...tableColumnSettings,
+    ...produce(tableColumnSettings, draft => {
```

Comment:
AFAIU the bug was caused by this file just using the default table column settings [that filters out](https://github.com/metabase/metabase/blob/94b298268465e4446f8c1fe17772e89259522063/frontend/src/metabase/visualizations/lib/settings/column.js#L536) fields with a visibility field of 'details-only;